bridge-token-factory/src/fee.rs

```diff
@@ -0,0 +1,661 @@
+use crate::*;
+
+fn adjust_fee_amount_between_bounds(fee_amount: u128, fee_bounds: &FeeBounds) -> u128 {
+    // If some token fee-bounds is set for deposit or withdraw than update the fee amount according to bounds
+    if let Some(lower_bound) = fee_bounds.lower_bound {
+        if fee_amount < lower_bound.0 {
+            return lower_bound.0;
+        }
+    }
+
+    if let Some(upper_bound) = fee_bounds.upper_bound {
+        if fee_amount > upper_bound.0 {
+            return upper_bound.0;
+        }
+    }
+
+    fee_amount
+}
+
+#[near_bindgen]
+impl BridgeTokenFactory {
+    //this should be added as per: 10% -> 0.1 = 0.1*10^6
+    #[access_control_any(roles(Role::FeeSetter))]
+    #[payable]
+    pub fn set_deposit_fee(
+        &mut self,
+        token: EthAddressHex,
+        fee_percentage: U128,
+        lower_bound: Option<U128>,
+        upper_bound: Option<U128>,
+    ) -> Promise {
+        self.deposit_fee.insert(
+            &token.0,
+            &Fee {
+                fee_percentage,
+                bounds: FeeBounds {
+                    lower_bound,
+                    upper_bound,
+                },
+            },
+        );
+
+        require!(env::attached_deposit() >= self.bridge_token_storage_deposit_required);
+        ext_bridge_token::ext(self.get_bridge_token_account_id(&token))
+            .with_static_gas(TOKEN_STORAGE_DEPOSIT_GAS)
+            .with_attached_deposit(env::attached_deposit())
+            .storage_deposit(None, None)
```

Comment:
What if we already have a registered token and we just want to update the data. We should prohibit update data or check if the token 

 Then either you need to explicitly prohibit updating the data, or check if the our account already registered for this token. 
 
 Do we refund excess deposit in this implementation?

Reply:
Actually, there is a refund, but the problem is that the refund is done to the predecessor of the `storage_deposit` call, which is the factory contract.
I will try to figure out a solution for this.

Reply:
Removed here bc39b1f074f2c9e0dfc2d3d0e76842ccd0b39e4b, because the storage deposit should be done manually while migration